geopandas/geoseries.py

```diff
@@ -299,16 +297,6 @@ def fillna(self, value=None, method=None, inplace=False,
         return super(GeoSeries, self).fillna(value=value, method=method,
                                              inplace=inplace, **kwargs)
 
-    def align(self, other, join='outer', level=None, copy=True,
-              fill_value=None, **kwargs):
-        if fill_value is None:
-            fill_value = BaseGeometry()
```

Comment:
Currently, GeoSeries.align is filling with empty geometries:

```
In [33]: s = geopandas.GeoSeries([Point(1, 1), Point(2, 2), Point(3, 3)])                                                                                     

In [34]: s[1:].align(s[:-1])                                                                                                                                  
Out[34]: 
(0    GEOMETRYCOLLECTION EMPTY
 1                 POINT (2 2)
 2                 POINT (3 3)
 dtype: object, 0                 POINT (1 1)
 1                 POINT (2 2)
 2    GEOMETRYCOLLECTION EMPTY
 dtype: object)
```

But pandas Series of GeoDataFrame uses NaNs:
```
In [35]: s2 = pd.Series(s)                                                                                                                                    

In [36]: s2[1:].align(s2[:-1])                                                                                                                                
Out[36]: 
(0            NaN
 1    POINT (2 2)
 2    POINT (3 3)
 dtype: object, 0    POINT (1 1)
 1    POINT (2 2)
 2            NaN
 dtype: object)

In [37]: df = geopandas.GeoDataFrame(geometry=s)                                                                                                              

In [38]: df[1:].align(df[:-1])                                                                                                                                
Out[38]: 
(      geometry
 0          NaN
 1  POINT (2 2)
 2  POINT (3 3),       geometry
 0  POINT (1 1)
 1  POINT (2 2)
 2          NaN)
```

Removing this implementation then uses the upstream one, getting the pandas behaviour for both GeoSeries and GeoDataFrame.

Reply:
A consequence of this alignment is that in binary operations, you currently get:

```
In [40]: s_poly = s.buffer(2)                                                                                                                                 

In [41]: s_poly[1:].intersection(s_poly[:-1])                                                                                                                 
Out[41]: 
0                                                   ()
1    POLYGON ((4 2, 3.990369453344394 1.80396571934...
2                                                   ()
dtype: object
```

where the `()` is the repr for an empty geometrycollection. On this branch, those would be missing values intead of empty geometries. This has an impact on methods like `area` (where empty gives 0, while missing would propagate as missing value in the result of `area`), but you could also argue this is more correct.

Reply:
> you could also argue this is more correct.

I think that it is more correct.